### ISSUE TITLE
Installation failed on Mac M1/M2

### ISSUE THREAD

**Issue by jinserk** [kserve/kserve#2852]
/kind feature

**Describe the solution you'd like**
Current Kserve v0.10.1 failed to be installed on Mac M1/M2 due to the `ray[serve]==2.0.0` dependency:
```
#27 37.11 Requirement already satisfied: psutil>=5.9.0 in /usr/local/lib/python3.10/site-packages (from kserve>=0.10.1->kai==0.5.9) (5.9.5)
#27 37.18 INFO: pip is looking at multiple versions of kserve to determine which version is compatible with other requirements. This could take a while.
#27 37.18 ERROR: Could not find a version that satisfies the requirement ray[serve]==2.0.0 (from kserve) (from versions: 2.2.0, 2.3.0, 2.3.1)
#27 37.18 ERROR: No matching distribution found for ray[serve]==2.0.0
```
We need to version of the ray up at least to 2.2.0 or higher.

**Comment by yuzisun** [comment]
@sivanantha321 i think we can make Ray as optional dependency ?

**Comment by sivanantha321** [comment]
> @sivanantha321 i think we can make Ray as optional dependency ?

yeah, sure !

**Comment by MikhailKravets** [comment]
The same problem occurs on Windows btw

**Comment by Nagarajj** [comment]
@sivanantha321 @yuzisun any ETA on this ? Also will there be a patch release with this fix ? Local testing with kserve 0.10.1 is not possible on M1 Mac due to this. Thanks.

**Comment by sivanantha321** [comment]
The ray serve is [upgraded to version 2.4](https://github.com/kserve/kserve/blob/97bada50b5b3abe32b0dc618560735017ec2a070/python/kserve/pyproject.toml#L45). Ray added support for Mac M1 chips from version 2.2. So the kserve installation should be fine from the next release 0.11.0
